bugbot/rules/webcompat_sightline.py

```diff
@@ -30,47 +30,44 @@ def handle_bug(
         bug_id = str(bug["id"])
         whiteboard = bug["whiteboard"]
 
-        if bug["id"] in self.sightline_ids:
-            if self.WHITEBOARD_ENTRY not in whiteboard:
-                self.autofix_changes[bug_id] = {
-                    "whiteboard": whiteboard + self.WHITEBOARD_ENTRY
-                }
-                return bug
-        else:
-            if self.WHITEBOARD_ENTRY in whiteboard:
-                self.autofix_changes[bug_id] = {
-                    "whiteboard": whiteboard.replace(self.WHITEBOARD_ENTRY, "")
-                }
-                return bug
+        if bug["id"] not in self.sightline_ids:
+            return None
+
+        is_sightline = self.sightline_ids[bug["id"]]
+        has_whiteboard = self.WHITEBOARD_ENTRY in whiteboard
+        if is_sightline and not has_whiteboard:
+            self.autofix_changes[bug_id] = {
+                "whiteboard": whiteboard + self.WHITEBOARD_ENTRY
+            }
+            return bug
+        if not is_sightline and has_whiteboard:
+            self.autofix_changes[bug_id] = {
+                "whiteboard": whiteboard.replace(self.WHITEBOARD_ENTRY, "")
+            }
+            return bug
 
         return None
 
     def get_bz_params(self, date) -> dict[str, Any]:
         fields = ["id", "summary", "whiteboard"]
         self.sightline_ids = self.get_sightline_bug_ids()
-        # Get all bugs that either have, or should have, the [webcompat:sightline]
-        # whiteboard entry
-        return {
-            "include_fields": fields,
-            "j_top": "OR",
-            "f1": "bug_id",
-            "o1": "anyexact",
-            "v1": ",".join(str(item) for item in self.sightline_ids),
-            "f2": "status_whiteboard",
-            "o2": "substring",
-            "v2": self.WHITEBOARD_ENTRY,
-        }
+        return {"include_fields": fields, "id": list(self.sightline_ids.keys())}
+        return fields
 
-    def get_sightline_bug_ids(self) -> set[int]:
+    def get_sightline_bug_ids(self) -> Mapping[int, bool]:
         project = "moz-fx-dev-dschubert-wckb"
         dataset = "webcompat_knowledge_base"
 
         client = gcp.get_bigquery_client(project, ["cloud-platform", "drive"])
         query = f"""
-        SELECT number FROM `{project}.{dataset}.webcompat_topline_metric_site_reports` as bugs
+        SELECT number, bugs.is_sightline FROM `{project}.{dataset}.scored_site_reports` as bugs
+        WHERE (bugs.is_sightline AND NOT CONTAINS_SUBSTR(bugs.whiteboard, "{self.WHITEBOARD_ENTRY}"))
+          OR (NOT bugs.is_sightline AND CONTAINS_SUBSTR(bugs.whiteboard, "{self.WHITEBOARD_ENTRY}"))
         """
 
-        return {row["number"] for row in client.query(query).result()}
+        return {
+            row["number"]: row["is_sightline"] for row in client.query(query).result()
```

Comment:
With this change, the function name became somewhat confusing. Now, it returns a dictionary with a boolean indicating whether the bug is a sightline, so the name `get_sightline_bug_ids()` no longer fits very well.

Reply:
Updated with `s/_ids/_data/` here and in the attribute name.